File: scrimage-core/src/test/scala/com/sksamuel/scrimage/Issue144Test.scala
package com.sksamuel.scrimage

import com.sksamuel.scrimage.color.Color
import org.scalatest.{Matchers, WordSpec}

class Issue144Test extends WordSpec with Matchers {

  "Cover resize" should {
    "not produce a black bar on an image with an odd size" in {
      val img = ImmutableImage.filled(2357, 2400, Color.White)

      val resized = img.cover(200, 200)
      val pixel   = resized.pixel(199, 0)

      pixel.red shouldBe 255
      pixel.blue shouldBe 255
      pixel.green shouldBe 255
    }
  }

}

File: scrimage-core/src/test/kotlin/com/sksamuel/scrimage/core/Issue144Test.kt
package com.sksamuel.scrimage.core

import com.sksamuel.scrimage.ImmutableImage
import io.kotest.core.spec.style.WordSpec
import io.kotest.matchers.shouldBe
import java.awt.Color

class Issue144Test : WordSpec({

   "Cover resize" should {
      "not produce a black bar on an image with an odd size" {
         val img = ImmutableImage.filled(2357, 2400, Color.WHITE)

         val resized = img.cover(200, 200)
         val pixel = resized.pixel(199, 0)

         pixel.red() shouldBe 255
         pixel.blue() shouldBe 255
         pixel.green() shouldBe 255
      }
   }

})